comment/views/comments.py

```diff
@@ -71,20 +110,20 @@ def dispatch(self, request, *args, **kwargs):
 
     def get(self, request, *args, **kwargs):
         context = self.get_context_data()
-        context['comment_form'] = CommentForm(instance=self.comment)
+        context['comment_form'] = CommentForm(instance=self.comment, request=self.request)
         context['comment'] = self.comment
         return render(request, 'comment/comments/update_comment.html', context)
 
     def post(self, request, *args, **kwargs):
-        form = CommentForm(request.POST, instance=self.comment)
+        form = CommentForm(request.POST, instance=self.comment, request=self.request)
         context = self.get_context_data()
         if form.is_valid():
             form.save()
             context['comment'] = self.comment
             return render(request, 'comment/comments/comment_content.html', context)
 
 
-class DeleteComment(BaseCommentView):
+class DeleteComment(BaseCommentMixin, AnonymousPermissionMixin, BaseCommentView):
```

Comment:
Same case for update class. I will move these permission checks from dispatch method to the separated permission class after this PR is merged